src/hooks/useOrderId.ts

```diff
@@ -0,0 +1,29 @@
+import { useHistory, useParams, useRouteMatch } from 'react-router'
+
+/**
+ * Fetches order id from url parameters
+ * Sanitizes it and updates the URL, if any
+ *
+ * @returns Sanitized order id
+ */
+export function useOrderId(): string {
+  const { orderId } = useParams<{ orderId: string }>()
+  const { url } = useRouteMatch()
+  const history = useHistory()
+
+  // Allows any kind of crap in the orderId, as long as there is a valid and continuous order id in it
+  // Ignores case
+  const regexMatch = orderId.match(/(0x[0-9a-f]{112})/i)
+
+  // Get extracted order id from the match, if any
+  const sanitizedOrderId = (regexMatch && regexMatch[0]?.toLowerCase()) || orderId.toLowerCase()
+
+  // If there's an orderId in the url AND it's not yet nice, update the URL
+  if (regexMatch && sanitizedOrderId !== orderId) {
+    // TODO: is there a better way to do this?
+    const newPath = url.replace(orderId, sanitizedOrderId)
```

Comment:
Im sure there is, you already decided for a parameter name `orderId`, so it would be to update the params by the parameter name. Not sure exactly how is done, but worse case you can get all the params, and merge the 2 objects.

Reply:
The other option I tried was to replace the path, which is `:orderId`
But that mean that if we ever change the route, we'll have to update it here too.
This hacky way works regardless of the path, replacing the old value with the new one.